File: examples/old_calculation_comparison/GASPROP3.m
function out=GASPROP3(T,property,gas,Prop,PropData)
%#ok<*OR2>
%#ok<*AND2>
%#ok<*UNRCH>
%#ok<*NODEF>

col=find(strcmp(Prop,property));
if strcmp(gas,'air')
    out=interp1(PropData(1:35,1),PropData(1:35,col),T); 
elseif strcmp(gas,'he')
    out=interp1(PropData(36:316,1),PropData(36:316,col),T); 
elseif strcmp(gas,'ar')
    out=interp1(PropData(317:347,1),PropData(317:347,col),T); 
elseif strcmp(gas,'co2')
    out=interp1(PropData(348:383,1),PropData(348:383,col),T); 
elseif strcmp(gas,'h2o')
    out=interp1(PropData(384:403,1),PropData(384:403,col),T); 
end

File: examples/old_calculation_comparison/gasprop3.py
from scipy.io import loadmat
from scipy.interpolate import interp1d
import pandas as pd
import numpy as np
from typing import Union

DATA = loadmat('GASPROPTABLE3')
DF = pd.DataFrame(DATA['PropData'], columns = [p[0] for p in DATA['Prop'][0]])

gas_row_index = {'air':[0,35],
                'he': [35,316],
                'ar':[316,347],
                'co2':[347,383], 
                'h2o':[383,-1]}

def get_property(T: Union[np.ndarray,float],
                prop: str,
                gas: str):

    """
    get the property specified by property of the gas specified by "gas" at the temperature specified by "T"
    """

    gas = gas.lower()
    if gas not in gas_row_index:
        raise ValueError('You have specified a gas that is not in the data base')

    return interp1d(DF['T'][gas_row_index[gas][0]:gas_row_index[gas][1]],
                    DF[prop][gas_row_index[gas][0]:gas_row_index[gas][1]])(T)

